Kconfig.zephyr

```diff
@@ -738,6 +738,40 @@ endmenu
 
 menu "Boot Options"
 
+config APPLICATION_VERSION_SUPPORTED
+	bool
+	help
+	  Hidden option used to indicate that the current image supports
+	  version information which may be embedded into the application or
+	  used for signing.
+
+if APPLICATION_VERSION_SUPPORTED
+
+config APPLICATION_VERSION_TYPE_MCUBOOT
+	bool
+	help
+	  Hidden option used to select mcuboot-style default version string of
+	  0.0.0+0
+
+config APPLICATION_VERSION_TYPE_INTEGER
+	bool
+	help
+	  Hidden option used to select integer-style default version string of
+	  0
+
+config APPLICATION_VERSION
```

Comment:
This is named application version and thus its format should be defined by Zephyr, but in reality when using MCUboot and imgtool then the format is fixed at `x.y.z-<build>` format.

This is bad, as user cannot decide an application format similar to the format used by the Zephyr kernel and have that present in generated `app_version.h`.
Instead users are forced to follow the imgtool format.

Zephyr community should define how app version format should be, and not just inherit imgtool's version format because that happens to used by a subtool.

imgtool tool version field should have it's own setting, and that setting can then default to be identical to the application version when possible.

Reply:
The version would be determined by the signing tool, in this case imgtool, if no signing tool is used then they're free to pick whatever they want, or if a signing tool which only supports a single integer version then that format. Note that the mcuboot version is completely compliant with semantic versioning whereby the extra version field is optional and not mandatory to be supported (and actually, it specifically states that that field should be discarded and ignored for determining version prescience).

The version field should be user selected to meet their needs, forcing a user to use one type of version without changing it is not user friendly and not going to be useful for users with alternative bootloaders.

Reply:
https://semver.org/

Reply:
> https://semver.org/

Indeed:

> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version
> Build metadata MAY be denoted by appending a plus sign and a series of dot separated identifiers immediately following the patch or pre-release version

Both are optional and do not need to be supported.